tests/pytests/unit/states/test_kubernetes.py

```diff
@@ -0,0 +1,811 @@
+    for key, value in secret_data["data"].items():
+        secret_data["data"][key] = base64.b64encode(
+            salt.utils.stringutils.to_bytes(value)
+        )
+
+    return secret_data
+
+
+def make_node_labels(name="minikube"):
+    return {
+        "kubernetes.io/hostname": name,
+        "beta.kubernetes.io/os": "linux",
+        "beta.kubernetes.io/arch": "amd64",
+        "failure-domain.beta.kubernetes.io/region": "us-west-1",
+    }
+
+
+def make_node(name="minikube"):
+    node_data = make_ret_dict(kind="Node", name="minikube")
+    node_data.update(
+        {
+            "api_version": "v1",
+            "kind": "Node",
+            "metadata": {
+                "annotations": {"node.alpha.kubernetes.io/ttl": "0"},
+                "labels": make_node_labels(name=name),
+                "name": name,
+                "namespace": None,
+                "link": "/api/v1/nodes/{name}".format(name=name),
+                "uid": "7811b8ae-c1a1-11e7-a55a-0800279fb61e",
+            },
+            "spec": {"external_id": name},
+            "status": {},
+        }
+    )
+    return node_data
+
+
+def make_namespace(name="default"):
+    namespace_data = make_ret_dict(kind="Namespace", name=name)
+    del namespace_data["data"]
+    namespace_data.update(
+        {
+            "status": {"phase": "Active"},
+            "spec": {"finalizers": ["kubernetes"]},
+            "metadata": {
+                "name": name,
+                "namespace": None,
+                "labels": None,
+                "link": "/api/v1/namespaces/{namespace}".format(
+                    namespace=name,
+                ),
+                "annotations": None,
+                "uid": "752fceeb-c1a1-11e7-a55a-0800279fb61e",
+            },
+        }
+    )
+    return namespace_data
+
+
+def make_ret_dict(kind, name, namespace=None, data=None):
+    """
+    Make a minimal example configmap or secret for using in mocks
+    """
+
+    assert kind in ("Secret", "ConfigMap", "Namespace", "Node")
+
+    if data is None:
+        data = {}
+
+    link = "/api/v1/namespaces/{namespace}/{kind}s/{name}".format(
+        namespace=namespace,
+        kind=kind.lower(),
+        name=name,
+    )
+
+    return_data = {
+        "kind": kind,
+        "data": data,
+        "api_version": "v1",
+        "metadata": {
+            "name": name,
+            "labels": None,
+            "namespace": namespace,
+            "link": link,
+            "annotations": {"kubernetes.io/change-cause": "salt-call state.apply"},
+        },
+    }
+    return return_data
+
+
+def test_configmap_present__fail():
+    error = kubernetes.configmap_present(
+        name="testme",
+        data={1: 1},
+        source="salt://beyond/oblivion.jinja",
+    )
+    assert {
+        "changes": {},
+        "result": False,
+        "name": "testme",
+        "comment": "'source' cannot be used in combination with 'data'",
+    } == error
```

Comment:
```suggestion
    assert error == {
        "changes": {},
        "result": False,
        "name": "testme",
        "comment": "'source' cannot be used in combination with 'data'",
    }
```
Always assert what we get against what we expect...

There are other assertions like this in the module.